cpp/src/cylon/table.cpp

```diff
@@ -1513,12 +1520,17 @@ Status DistributedEquals(const std::shared_ptr<cylon::Table> &a,
     RETURN_CYLON_STATUS_IF_FAILED(Equals(a, b_repartitioned, subResult, true));
   }
 
-  RETURN_CYLON_STATUS_IF_FAILED(mpi::AllReduce(a->GetContext(),
-                                               &subResult,
-                                               &result,
-                                               1,
-                                               Bool(),
-                                               cylon::net::LAND));
+  auto sub_result_scalar =
+      std::make_shared<Scalar>(std::move(arrow::MakeScalar(int8_t(subResult))));
```

Comment:
when you are passing a `shared_ptr` to a constructor (Rvalue), moving is unnecessary. you would only have to move Lvalues. 
```suggestion
  auto sub_result_scalar = std::make_shared<Scalar>(arrow::MakeScalar(int8_t(subResult)));
```